### ISSUE TITLE
Tracked changes: investigation, possibility to reproduce in Apache FOP

### ISSUE THREAD

**Issue by Intelligent2013** [metanorma/mn2pdf#64]
1. Addition/deletion changes are shown in red and green.  Example document: https://shop.bsigroup.com/ProductDetail?pid=000000000030413431

![image](https://user-images.githubusercontent.com/5559926/115211321-1e97b080-a108-11eb-8acb-a8ec6cec0bba.png)

![image](https://user-images.githubusercontent.com/5559926/115211400-33744400-a108-11eb-861e-4cb1f01d8853.png)

Supported yet:
1.1. Added text are shown in green color background - tag `<add>...</add>`, the style (background's color and text's color can be set up in common.xsl)
1.2. Deleted text are shown in red color text - tag `<del>...</del>`.

To be check:
1.3. Added figure are shown in the green box.
1.4. Deleted figure are shown in the red box with cross.
1.5. Added table are shown in the green box.
1,6. Deleted table are shown in the red box with cross.

2. The additions are tagged with symbols. Example document: https://civilnode.com/download-standard/10646400567105/bs-8500-1-concrete-complementary-british-standard-to-bs-en-206-part-1-method-of-specifying

![image](https://user-images.githubusercontent.com/5559926/115212264-fb213580-a108-11eb-9f77-c07d9d3913f5.png)

To be check:
2.1. SVG graphics with dynamic text (for tags) inlined in the text.

**Comment by Intelligent2013** [comment]
Proposals for MN XML markup:
1. `<add>....</add>` - all markup inside these tags will be showed as added thing (green background text, green bordered figure/table)
2. element level track changes, for example `<table track_changes="add">...</table>`  - table, marked with attribute `track_changes="add"` will be showed as 'added' (with green border).
3. clause level track changes, for example `<clause  track_changes="add">...</clause`> - all markup inside clause with attribute  `track_changes="add"` will be showed as added thing.
4. annex level track changes - similar to 'clause' level track changes.

**Comment by Intelligent2013** [comment]
> 1.3. Added figure are shown in the green box

xslt updated for support 'added' figure. Example:
![image](https://user-images.githubusercontent.com/5559926/115397278-97713800-a1ee-11eb-80ca-a3981056b98f.png)


> 1.4. Deleted figure are shown in the red box with cross.

xslt updated for support 'deleted' figure. Example:
![image](https://user-images.githubusercontent.com/5559926/115398461-db187180-a1ef-11eb-9e88-f198de305c2a.png)

(I can't fit 'red crossing'  exactly to the border and figure title.)

> 1.5. Added table are shown in the green box.

xslt updated for support 'added' table. Example:
![image](https://user-images.githubusercontent.com/5559926/115399321-bcff4100-a1f0-11eb-8426-4f53617c9572.png)

> 1,6. Deleted table are shown in the red box with cross.

xslt updated for support 'deleted' table, but I can't display red crossing over table. I propose to display 'deleted' table with 'light red' background with red border:
![image](https://user-images.githubusercontent.com/5559926/115399855-49a9ff00-a1f1-11eb-8d42-09d0b79eaab7.png)


or with red border and all text in red.

p.s. xslt should be update later for actual xml element/attribute for added/deleted indication.

**Comment by Intelligent2013** [comment]
> 2.1. SVG graphics with dynamic text (for tags) inlined in the text.

xslt updated. For example source xml (proof-of-concept, just for testing):
```
<p>
<del>The use of echo cancellers on the VBD channel, as per Rec. ITU-T G.168.</del><add amendment="2">ADDED: The use of echo cancellers on the VBD channel, as per Rec. ITU-T G.168.</add>
</p>
<p>
<add corrigenda="1">ADDED: The use of echo cancellers on the VBD channel, as per Rec. ITU-T G.168.</add>
</p>
```

![image](https://user-images.githubusercontent.com/5559926/115443086-7d9a1a00-a21b-11eb-98e1-6d59dabb30af.png)

Xslt should be changed later for actual xml structures.

**Comment by ronaldtse** [comment]
This looks great. So we will need help from @opoudjis to update the XML and make this work!

**Comment by ronaldtse** [comment]
> xslt updated for support 'deleted' table, but I can't display red crossing over table. I propose to display 'deleted' table with 'light red' background with red border:

@Intelligent2013 I agree, light red background looks more pleasing. Let's go with that.

**Comment by opoudjis** [comment]
> This looks great. So we will need help from @opoudjis to update the XML and make this work!

We do? This is already implemented, albeit at inline level.
